addons/account/views/account_invoice_view.xml

```diff
@@ -432,16 +450,40 @@
                     <field name="sent" invisible="1"/>
                     <notebook colspan="4">
                         <page string="Invoice Lines">
-                            <field name="invoice_line_ids" nolabel="1" widget="one2many_list" mode="tree,kanban" context="{'type': type, 'journal_id': journal_id, 'default_invoice_id': id}">
+                            <field
+                                name="invoice_line_ids"
+                                nolabel="1"
+                                widget="section-and-note"
+                                mode="tree,kanban"
+                                context="{'type': type, 'journal_id': journal_id, 'default_invoice_id': id}"
+                            >
                                 <tree string="Invoice Lines" editable="bottom">
+                                    <!-- if it is defined, it overrides the default button -->
+                                    <control>
+                                        <create string="Add a line" context="{'default_line_type': 'line_accountable'}" />
+                                        <create string="Add a section" context="{'default_line_type': 'line_section'}" always-show-dialog="1" />
+                                        <create string="Add a note" context="{'default_line_type': 'line_note'}" always-show-dialog="1" />
+                                    </control>
+
                                     <field name="sequence" widget="handle"/>
                                     <field name="product_id" domain="[('sale_ok','=',True)]"/>
                                     <field name="origin" invisible="1"/>
                                     <field name="is_rounding_line" invisible="1"/>
-                                    <field name="name"/>
+                                    <field name="name" class="product_name"/>
+                                    <field name="line_type" invisible="1"/>
                                     <field name="company_id" invisible="1"/>
-                                    <field name="account_id" groups="account.group_account_user"
-                                        domain="[('company_id', '=', parent.company_id), ('internal_type', '=', 'other'), ('deprecated', '=', False)]"/>
+                                    <field
```

Comment:
Do we really need 12 lines for 1 field ?

Reply:
Isn't it better this way than having an hardly readable line of 700 characters? What's a good compromise?